sound/soc/sof/compress.c

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause)
```

Comment:
you should not make the simplification that compress == probe. Probes rely on the compressed streams, but you can send a compressed stream which is NOT a probe stream.